geopmdpy/install_user.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+#  Copyright (c) 2015 - 2024 Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+# A script to install geopmdpy based on latest upstream dev branch
+#
+# Note: Will clone geopm into current working directory if geopm
+#       directory does not exist
+
+set -e
+if [ -z "$INSTALL_PREFIX" ]; then
+    echo "HELP: To select a different install location export INSTALL_PREFIX"
+    INSTALL_PREFIX=$HOME/geopm-build
+fi
+if [ -z "$MPI_DIR" ]; then
+    echo "HELP: To select a different MPI directory location export MPI_DIR"
+    MPI_DIR=$(dirname $(dirname $(which mpicc)))
+fi
+echo "INSTALL_PREFIX=$INSTALL_PREFIX"
+echo "MPI_DIR=$MPI_DIR"
+set -x
+sleep 2
+echo "[mpi]
+mpi_dir = $MPI_DIR
+mpicc   = %(mpi_dir)s/bin/mpicc
+mpicxx  = %(mpi_dir)s/bin/mpicxx
+include_dirs         = %(mpi_dir)s/include
+libraries            = mpi
+library_dirs         = %(mpi_dir)s/lib
+runtime_library_dirs = %(library_dirs)s" > mpi.cfg
+MPI4PY_BUILD_MPICFG=$PWD/mpi.cfg python3 -m pip install mpi4py
+if [ ! -d geopm ]; then
+    git clone https://github.com/geopm/geopm.git
+fi
+cd geopm
+git checkout dev
+git fetch origin
+git merge --ff-only origin/dev
+cd libgeopmd
+./autogen.sh
+./configure --enable-levelzero --prefix=$INSTALL_PREFIX
+make -j20
+make install
+export C_INCLUDE_PATH="$INSTALL_PREFIX/include"
+export LIBRARY_PATH="$INSTALL_PREFIX/lib:$INSTALL_PREFIX/lib64"
+cd ../geopmdpy
+./make_sdist.sh
```

Comment:
When make_sdist.py executes, it relies on ``import build`` which is specified in the requirements.txt for geopmdpy.

If this script is meant to be an easy button for end users to install from dev, it should also install the Python requirements that are needed for geopmdpy.  We've had users hit this issue recently.

Additionally, perhaps it would be better to root cause what's going on w/mcfly and see if this call to make_sdist.sh can be avoided entirely.  You'd still need to install the dev python requirements for geopmdpy (assuming they are different from whatever is installed, if GEOPM had previously been installed).

IMO this should all be handled through spack instead of a special script.  That's the proper way to handle this dependency on mpi4py.

Also, I'm wondering if this belongs in ``<REPO_ROOT>/integration/config`` instead of the spot it is in now?  This is just a helper script for building dev.  It won't be used for tagged releases.  Not sure it should be featured so prominently in the geopmdpy root dir.